src/resources/Catalogs/tests/ProductListingConfiguration.spec.ts

```diff
@@ -0,0 +1,153 @@
+import API from '../../../APICore';
+import {FieldOperatorType, FieldValueType} from '../../Enums';
+import {New} from '../../BaseInterfaces';
+import ProductListingConfiguration from '../ProductListingConfiguration';
+import {ProductListingConfigurationModel, RankingTypeEnum} from '../ProductListingConfigurationInterfaces';
+
+jest.mock('../../../APICore');
+
+const APIMock: jest.Mock<API> = API as any;
+
+describe('ProductListingConfiguration', () => {
+    const catalogId = 'catalogId';
+    const productListingId = 'productlistingId';
+    let productListingConfiguration: ProductListingConfiguration;
+    const api = new APIMock() as jest.Mocked<API>;
+    const serverlessApi = new APIMock() as jest.Mocked<API>;
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        productListingConfiguration = new ProductListingConfiguration(api, serverlessApi);
+    });
+
+    describe('list', () => {
+        it('should make a GET call to the product listing configuration base url', () => {
+            productListingConfiguration.list(catalogId, productListingId);
+            expect(api.get).toHaveBeenCalledTimes(1);
+            expect(api.get).toHaveBeenCalledWith(
+                `${ProductListingConfiguration.baseUrl}/${catalogId}/productlistings/${productListingId}/configurations`
+            );
+        });
+    });
+
+    describe('create', () => {
+        it('should make a POST call to the product listing configuration base url', () => {
+            const productListingConfigurationModel: New<ProductListingConfigurationModel> = {
+                rankingConfigurations: [
+                    {
+                        include: [
+                            {
+                                fieldName: 'ec_brand',
+                                operator: FieldOperatorType.IS_EXACTLY,
+                                value: {
+                                    type: 'STRING',
+                                    value: FieldValueType.STRING,
+                                },
+                            },
+                        ],
+                        exclude: [],
+                        type: RankingTypeEnum.BURY,
+                        value: 10,
+                    },
+                    {
+                        include: [
+                            {
+                                fieldName: 'ec_brand',
+                                operator: FieldOperatorType.IS_EXACTLY,
+                                value: {
+                                    type: 'string',
+                                    value: FieldValueType.STRING,
+                                },
+                            },
+                        ],
+                        exclude: [],
+                        type: RankingTypeEnum.BOOST,
+                        value: 15,
+                    },
+                ],
+            };
```

Comment:
nit-pic, feel free to ignore: looks like to me the two config model used in the tests are quite similar with the exception of the BOOST value.

Would it be possible to refactor them to eases the reading of the file

Reply:
Yes I like your idea